### PR TITLE
[DOC] Doc conversion and update for examples

### PR DESCRIPTION
**Pull request by outoftardis**
# Description

- Converted the remaining algorithms into RST:

  - [MSE](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/objective-functions/mse.html)
  - [Log Loss](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/objective-functions/logistic-loss.html)
  - [Cross Entropy](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/objective-functions/cross-entropy.html)
  - [SAGA](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/solvers/stochastic-average-gradient-accelerated-method.html)
  - [CDA](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/solvers/coordinate-descent.html)
  - [online](https://outoftardis.github.io/daal/dev_guide/algorithms/pca/computation-online.html) and [distributed](https://outoftardis.github.io/daal/dev_guide/algorithms/pca/computation-distributed.html) PCA: 
  - [AdaBoost MultiClass](https://outoftardis.github.io/daal/dev_guide/algorithms/boosting/adaboost-multiclass.html)
  - distributed Implicit [ALS](https://outoftardis.github.io/daal/dev_guide/algorithms/implicit_als/implicit-alternating-least-squares.html): 
    - [initialization](https://outoftardis.github.io/daal/dev_guide/algorithms/implicit_als/initialization-distributed.html)
    - [computation: training](https://outoftardis.github.io/daal/dev_guide/algorithms/implicit_als/computation-distributed-training.html#implicit-als-distributed-training)
    - [computation: prediction](https://outoftardis.github.io/daal/dev_guide/algorithms/implicit_als/computation-distributed-prediction.html#implicit-als-distributed-prediction)

- Fixed broken links in docs
- Updated the script that creates rst files for examples. Removed the links to the cpp_sycl examples, added oneAPI examples for DPC++ and C++.
- Added links to daal4py examples
- Added a note about Java and GPU

[Output](https://outoftardis.github.io/daal/)

### CODE REVIEW

**Comment by averbukh** [comment]
I see a lot of not rendered tags in the rich view mode. It's wrong rst tags or github rst render issue. If it's latter, do you have rendered content for that?

**Comment by outoftardis** [comment]
> I see a lot of not rendered tags in the rich view mode. It's wrong rst tags or github rst render issue. If it's latter, do you have rendered content for that?

See the artifact for this PR: https://dev.azure.com/daal/DAAL/_build/results?buildId=3494&view=artifacts&type=publishedArtifacts

**Comment by outoftardis** [comment]
@averbukh I've updated gh-pages in my fork with the output for this PR.

To view the rendered output for the converted topics, see:

- [MSE](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/objective-functions/mse.html)
- [Log Loss](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/objective-functions/logistic-loss.html)
- [Cross Entropy](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/objective-functions/cross-entropy.html)
- [SAGA](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/solvers/stochastic-average-gradient-accelerated-method.html)
- [CDA](https://outoftardis.github.io/daal/dev_guide/algorithms/optimization-solvers/solvers/coordinate-descent.html)
- [online](https://outoftardis.github.io/daal/dev_guide/algorithms/pca/computation-online.html) and [distributed](https://outoftardis.github.io/daal/dev_guide/algorithms/pca/computation-distributed.html) PCA: 
- [AdaBoost MultiClass](https://outoftardis.github.io/daal/dev_guide/algorithms/boosting/adaboost-multiclass.html)
- distributed Implicit [ALS](https://outoftardis.github.io/daal/dev_guide/algorithms/implicit_als/implicit-alternating-least-squares.html): 
  - [initialization](https://outoftardis.github.io/daal/dev_guide/algorithms/implicit_als/initialization-distributed.html)
  - [computation: training](https://outoftardis.github.io/daal/dev_guide/algorithms/implicit_als/computation-distributed-training.html#implicit-als-distributed-training)
  - [computation: prediction](https://outoftardis.github.io/daal/dev_guide/algorithms/implicit_als/computation-distributed-prediction.html#implicit-als-distributed-prediction)

**Comment by napetrov** [comment]
Examples update looks good

File: docs/source/dev_guide/algorithms/svm/support-vector-machine-classifier.rst [hunk 1]
```diff
@@ -230,36 +232,49 @@ Examples
...
+    Batch Processing:
+    
+    - :ref:`dpc_svm_two_class_thunder_dense_batch.cpp`
+
+  .. tab:: oneAPI C++
 
     Batch Processing:
     
-    -  :ref:`svm_two_class_thunder_dense_batch.cpp`
+    - :ref:`cpp_svm_two_class_smo_dense_batch.cpp`
```
Comment (PetrovKP) [inline docs/source/dev_guide/algorithms/svm/support-vector-machine-classifier.rst:245]: oneAPI C++ have 2 examples for SVM (SMO & thunder method)
```
:ref:`cpp_svm_two_class_thunder_dense_batch.cpp
:ref:`cpp_svm_two_class_smo_dense_batch.cpp
```

File: docs/source/dev_guide/algorithms/logistic_regression/logistic-regression.rst [hunk 2]
```diff
@@ -261,27 +263,29 @@ Examples
...
     -  :java_example:`LogRegBinaryDenseBatch.java <logistic_regression/LogRegBinaryDenseBatch.java>`
 
+  .. tab:: Python* with DPC++ support
+
+    Batch Processing:
+
+    - :daal4py_sycl_example:`log_reg_dense_batch.py`
+    - :daal4py_sycl_example:`log_reg_binary_dense_batch.py`
+
   .. tab:: Python*
```
Comment (PetrovKP) [inline docs/source/dev_guide/algorithms/logistic_regression/logistic-regression.rst:290]: For python example missed log_reg_binary_dense_batch.py. See 294 line in this file

**Comment by PetrovKP** [comment]
I am confused by the different style of tables with parameters. In some places they are not stand out, in others they stand out. Is this expected?

For example:
Objective Function:
image

The sum of functions algorithm:
image

**Comment by outoftardis** [comment]
> I am confused by the different style of tables with parameters. In some places they are not stand out, in others they stand out. Is this expected?
> 
> For example:
> Objective Function:
> image
> 
> The sum of functions algorithm:
> image

This is expected, yes. I'm fixing it gradually.

File: docs/source/dev_guide/algorithms/implicit_als/computation-distributed-training.rst [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+.. contents::
+    :local:
+    :depth: 1
+
+Algorithm Parameters
+********************
+
+At the training stage, implicit ALS recommender in the distributed processing mode has the following parameters:
+
```
Comment (Vika-F) [inline docs/source/dev_guide/algorithms/implicit_als/computation-distributed-training.rst:32]: It seems the table with parameters is missing here.

File: docs/source/dev_guide/algorithms/implicit_als/computation-distributed-training.rst [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+The Input Of Step 3 From Init is a key-value data collection that refers to the ``outputOfInitForComputeStep3`` output of the initialization stage:
+
+- Output of the :ref:`step 1 of the distributed initialization algorithm <implicit_als_distributed_init_step_1>` in :ref:`part 1 <implicit_als_computation_parts>` of the iteration
+- Output of the :ref:`step 2 of the distributed initialization algorithm <implicit_als_distributed_init_step_2>` in :ref:`part 2 <implicit_als_computation_parts>` of the iteration
+
+.. image:: images/implicit-als-distributed-computation-training-step-3.png
+    :width: 600
+    :align: center
+
```
Comment (Vika-F) [inline docs/source/dev_guide/algorithms/implicit_als/computation-distributed-training.rst:167]: The text "In this step, implicit ALS recommender training accepts the input described below. ... " is missing here.

File: docs/source/dev_guide/algorithms/pca/computation-batch.rst [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+   * - Input ID
+     - Input
+   * - data
+     - Use when the input data is a normalized or non-normalized data set.
+       Pointer to the :math:`n \times p` numeric table that contains the input data set.
+       This input can be an object of any class derived from NumericTable.
+   * - correlation
+     - Use when the input data is a correlation matrix. Pointer to the :math:`p \times p`
+       numeric table that contains the correlation matrix. This input can be an
+       object of any class derived from NumericTable except
```
Comment (Vika-F) [inline docs/source/dev_guide/algorithms/pca/computation-batch.rst:43]: In ALS algorithm the text "This input can be an object of any class ..." is formatted as a Note statement. Should it be formatted the same way across all algorithms?

File: docs/source/dev_guide/algorithms/pca/computation-batch.rst [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+****************
+
+.. contents::
+    :local:
+    :depth: 1
+
+Algorithm Input
+---------------
+The PCA algorithm accepts the input described below. Pass the
+Input ID as a parameter to the methods that provide input for your algorithm.
```
Comment (Vika-F) [inline docs/source/dev_guide/algorithms/pca/computation-batch.rst:27]: "For more details, see Algorithms." is missing here.

File: docs/source/dev_guide/algorithms/pca/computation-batch.rst [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+   :align: left
+
+   * - Result ID
+     - Result
+   * - eigenvalues
+     - Pointer to the :math:`1 \times p_r` numeric table that contains eigenvalues
+       in the descending order. By default, this result is an object of the
+       HomogenNumericTable class, but you can define the result as an object of
+       any class derived from NumericTable except PackedSymmetricMatrix,
+       PackedTriangularMatrix, and CSRNumericTable.
```
Comment (Vika-F) [inline docs/source/dev_guide/algorithms/pca/computation-batch.rst:132]: In ALS algorithm the text "This input can be an object of any class ..." is formatted as a Note statement. Should it be formatted the same way across all algorithms?
The same for 'eigenvectors'.

File: docs/source/dev_guide/algorithms/svm/support-vector-machine-classifier.rst (hunk 1 above)
Comment (outoftardis) [reply docs/source/dev_guide/algorithms/svm/support-vector-machine-classifier.rst:245]: fixed

File: docs/source/dev_guide/algorithms/logistic_regression/logistic-regression.rst (hunk 2 above)
Comment (outoftardis) [reply docs/source/dev_guide/algorithms/logistic_regression/logistic-regression.rst:290]: fixed

File: docs/source/dev_guide/algorithms/pca/computation-batch.rst (hunk 6 above)
Comment (outoftardis) [reply docs/source/dev_guide/algorithms/pca/computation-batch.rst:27]: fixed

File: docs/source/dev_guide/algorithms/implicit_als/computation-distributed-training.rst (hunk 3 above)
Comment (outoftardis) [reply docs/source/dev_guide/algorithms/implicit_als/computation-distributed-training.rst:32]: fixed, thanks

File: docs/source/dev_guide/algorithms/implicit_als/computation-distributed-training.rst (hunk 4 above)
Comment (outoftardis) [reply docs/source/dev_guide/algorithms/implicit_als/computation-distributed-training.rst:167]: fixed, thanks

File: docs/source/dev_guide/algorithms/pca/computation-batch.rst (hunk 5 above)
Comment (outoftardis) [reply docs/source/dev_guide/algorithms/pca/computation-batch.rst:43]: fixed

File: docs/source/dev_guide/algorithms/pca/computation-batch.rst (hunk 7 above)
Comment (outoftardis) [reply docs/source/dev_guide/algorithms/pca/computation-batch.rst:132]: Yes, you are right. But at the moment there are still instances where this text isn't converted into a note, this is not the only one. I'm changing this gradually.

**Comment by outoftardis** [comment]
@PetrovKP @Vika-F I've addressed all of your comments.
There are still some inconsistencies in formatting in the doc, please ignore it for now. I'm aware that they exist and I'm working on fixing them.

**Comment by outoftardis** [comment]
@averbukh @Vika-F @PetrovKP @napetrov do you have any other comments regarding this PR?

**Comment by napetrov** [comment]
@outoftardis  could you please rebase so internal ci can validate content.
Overall i don't have other comments

**Comment by napetrov** [comment]
Looks good for me. @averbukh  - is there any questions/comments or we can go and merge?

**Review by napetrov — APPROVED** [review]
